pallets/asset-manager/src/lib.rs

```diff
@@ -283,6 +312,21 @@ pub mod pallet {
 			LocationAssetId::<T>::remove(&old_location);
 			LocationAssetId::<T>::insert(&location, &asset_id);
 			AssetIdLocation::<T>::insert(&asset_id, &location);
+
+			// extract para id from old multilocation and delete it.
+			if let Some(old_para_id) =
+				Self::get_para_id_from_multilocation(old_location.into().as_ref())
+			{
+				AllowedDestParaIds::<T>::remove(old_para_id);
+			}
```

Comment:
This is not correct yet. You can have mutliple assets for the same parachain id, You need to extract the `old_para_id` and iterate `AssetIdLocation` to check if there are more assets for that ID. If not, then remove the ID from `AllowedDestParaIds`

Reply:
Or we can make `AllowedDestParaIds` be map of `(ID, num_of_associated_assets)` . 
So when we remove an asset from `AssetIdLocation` we can reduce `num_of_associated_assets`. When we insert an asset we can increment `num_of_associated_assets`. Then we won't have to constantly iterate over `AssetIdLocaiton`

Reply:
The latter is probably better, but it can be left as improvement IMO, this PR is getting out of hand. Also iterating `AssetIdLocation` is not a huge deal as update happens rarely and controlled by sudo.

Reply:
Yeah, just change the storage to `StorageMap<ParachainID, uint32>`, the later is the count of parachain_id's appearance.